enzy_htp/analysis/residue_pka.py

```diff
@@ -0,0 +1,43 @@
+from typing import List
+from enzy_htp.core.logger import _LOGGER
+
+# Importing PROPKA components
+from propka.lib import loadOptions
+from propka.input import read_parameter_file, read_molecule_file
+from propka.parameters import Parameters
+from propka.molecular_container import MolecularContainer
+
+
+def get_residue_pka(target_mask: str, pdb_path: str) -> List[float]:
```

Comment:
I like the name of this API! Please also add the docstring mimiking stability.py or other APIs. While writing the docstring, you can think about these questions:

1. What is the most generic defination of this API. (hint: think about the generic input and output)
2. By the generic defination, what changes would be needed in this API in the future (e.g.: we may provide multiple engines such as constant pH MD, or machine learning based method for users to choose)
3. To make future changes smooth (i.e.: we dont need to write repetitive code, changing the code wont cause problems to existing code that used this API), what can we do to this API. (hint: reference APIs in protonation.py stability.py or dipole.py)